gerrit.go

```diff
@@ -265,18 +271,22 @@ func (c *Client) DeleteRequest(urlStr string, body interface{}) (*Response, erro
 	return c.Do(req, nil)
 }
 
-// RemoveMagicPrefixLine removed the "magic prefix line" of Gerris JSON response.
-// the JSON response body starts with a magic prefix line that must be stripped before feeding the rest of the response body to a JSON parser.
-// The reason for this is to prevent against Cross Site Script Inclusion (XSSI) attacks.
+// RemoveMagicPrefixLine removes the "magic prefix line" of Gerris JSON
+// response if present. The JSON response body starts with a magic prefix line
+// that must be stripped before feeding the rest of the response body to a JSON
+// parser. The reason for this is to prevent against Cross Site Script
+// Inclusion (XSSI) attacks.  By default all standard Gerrit APIs include this
+// prefix line though some plugins may not.
 //
 // Gerrit API docs: https://gerrit-review.googlesource.com/Documentation/rest-api.html#output
 func RemoveMagicPrefixLine(body []byte) []byte {
-	index := bytes.IndexByte(body, '\n')
-	if index > -1 {
-		// +1 to catch the \n as well
-		body = body[(index + 1):]
+	if bytes.HasPrefix(body, []byte(")]}'\n")) {
+		index := bytes.IndexByte(body, '\n')
+		if index > -1 {
+			// +1 to catch the \n as well
+			body = body[(index + 1):]
+		}
 	}
```

Comment:
This code can be significantly simplified without changing behavior.

If body has `")]}'\n"` prefix, there's no need to use `bytes.IndexByte` to find the first `'\n'` character, it's known to be at position 3 because `bytes.HasPrefix(body, []byte(")]}'\n"))` was true.

Also, since this is called often, it's probably a good idea to factor out `[]byte(")]}'\n")` into a single package scope variable, instead of potentially allocating once per `RemoveMagicPrefixLine` call.

```Go
func RemoveMagicPrefixLine(body []byte) []byte {
	if bytes.HasPrefix(body, magicPrefix) {
		return body[5:]
	}
	return body
}

var magicPrefix = []byte(")]}'\n")
```